tests/foreman/api/test_syncplan.py

```diff
@@ -1127,3 +1199,29 @@ def test_positive_delete_synced_product(self):
         sync_plan.delete()
         with self.assertRaises(HTTPError):
             sync_plan.read()
+
+    @run_only_on('sat')
+    @upgrade
+    def test_positive_delete_synced_product_custom_cron(self):
+        """Create a sync plan with custom cron with one synced
+        products and delete it.
+
+        :id: f13936f5-7522-43b8-a986-26795637cde9
+
+        :expectedresults: A sync plan is created with one synced product and
+            sync plan can be deleted.
+
+        :CaseLevel: Integration
+        """
+        sync_plan = entities.SyncPlan(
+            organization=self.org,
+            interval='custom cron',
+            cron_expression=gen_choice((valid_cron_expressions()))).create()
+        product = entities.Product(organization=self.org).create()
+        entities.Repository(product=product).create()
+        sync_plan.add_products(data={'product_ids': [product.id]})
+        product.sync()
+        sync_plan.delete()
+        self.assertEqual(None, product.sync_plan.id)
```

Comment:
is this really working without reading the product again, as per my understanding product.sync_plan should be None.
eg we have to read the product twice one time after associating the sync plan and assert that product sync plan exist, and one time after deleting the sync plan and assert that product sync plan does not exist, and after try to read the sync plan it self that should raise the expected HTTP error. 
what do you think?

Reply:
agreed, done!

Reply:
any way think there is a bug in nailgun product read  if sync_plan is null in server response we should not have sync_plan with id product sync plan must be None.

Reply:
updated this to product.sync_plan for now